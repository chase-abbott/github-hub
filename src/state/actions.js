export const INPUT = 'INPUT';
export const SUBMIT = 'SUBMIT';

export const inputAction = (value) => ({ type: INPUT, payload: value });
export const submitAction = (value) => ({ type: SUBMIT, payload: value });
