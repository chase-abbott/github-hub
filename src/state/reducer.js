import { INPUT, SUBMIT } from './actions';

export const initialState = {
  inputValue: '',
  repositories: []
};

export const reducer = (state, action) => {
  switch(action.type){
    case INPUT:
      return {
        ...state,
        inputValue: action.payload
      }; 
    case SUBMIT: 
      return {
        ...state,
        repositories: action.payload
      };
    default:
      return state;
  }
};
