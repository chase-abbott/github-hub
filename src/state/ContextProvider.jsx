/* eslint-disable react/prop-types */
import React, { useContext, createContext, useReducer } from 'react';
import { initialState, reducer } from './reducer';

const Context = createContext();

export const ContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>
      {children}
    </Context.Provider>
  );
};

export const useReducerState = () => {
  const { state } = useContext(Context);
  return state;
};

export const useDispatch = () => {
  const { dispatch } = useContext(Context);
  return dispatch;
};
