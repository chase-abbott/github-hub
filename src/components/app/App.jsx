import React from 'react';
import { useDispatch, useReducerState } from '../../state/ContextProvider';
import { inputAction } from '../../state/actions';

export default function App() {
  const dispatch = useDispatch();
  const state = useReducerState();
  return (
    <input 
      value={state.inputValue}
      onChange={({ target }) => dispatch(inputAction(target.value))}
    />
  );
}
