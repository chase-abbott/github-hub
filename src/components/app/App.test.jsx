/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { ApolloProvider, createHttpLink, ApolloClient, InMemoryCache } from '@apollo/client';
import { ContextProvider } from '../../state/ContextProvider';
import { setContext } from '@apollo/client/link/context';

const key = process.env.ACCESS_TOKEN;

const link = createHttpLink({
  uri: 'https://api.github.com/graphql'
});

const authLink = setContext((_, { headers }) => {
  return {
    headers: {
      ...headers,
      authorization: `Bearer ${key}`
    }
  };
});

const client = new ApolloClient({
  link: authLink.concat(link),
  cache: new InMemoryCache()
});

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <ApolloProvider client={client}>
        <ContextProvider>
          <App />
        </ContextProvider>
      </ApolloProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
