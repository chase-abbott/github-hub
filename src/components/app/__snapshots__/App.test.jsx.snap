// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; align-items: center; justify-content: center; margin-top: 25px;"
  >
    <label>
       GitHub Username: 
    </label>
    <form
      style="display: flex; flex-direction: column; align-items: center;"
    >
      <input
        value=""
      />
      <button>
         Search 
      </button>
    </form>
    <ul
      style="list-style: none;"
    />
  </div>
</DocumentFragment>
`;
